setup.md

```diff
@@ -0,0 +1,92 @@
+### Install mbed-cli to a venv
```

Comment:
Assuming these instructions are useful to have here, they aren't easy to follow. Most readers aren't Python experts. A typical reader's reaction here: what's a venv? why should I care? what's a virtual environment?

I recommend to NOT include virtualenv instructions here. Just do `pip install --user mbed-cli`. If you do, you need to explain that it's about Python and only about Python, that it matters for mbed-cli because it's written in Python, and that if the reader does this then they need to run `virtualenv venvs/mbed` to open a new shell from which they can then run `mbed`.

Reply:
mbed-cli is really horrible to your default python environment. It does way too much installing and even did version downgrading in the past (might still). Even though the title is "venv" and you may not have a clue what that is as a new developer, the why you'd want to is explained.

Reply:
I've changed this paragraph to have two sets of instructions - one without a venv, and a second - with.